### PR TITLE
Feat: set right default value for oneOf schema if set

### PR DESCRIPTION
**Pull request by chivalryq**
### Description of your changes

Fix the case that setting default value of basic types to fields with multi-types in CUE when generating SDK. The default value should belongs to the corresponding type of all candidates.  e.g. 
```cue
cpu?: *1 | number | string
```
In this example, default `1` should be the default value when CPU is `number` type.

Fix: https://github.com/kubevela/kubevela/actions/runs/4400853066/jobs/7706538333 

cc @StevenLeiZhang 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/definition/gen_sdk/gen_sdk.go [hunk 1]
```diff
@@ -414,10 +419,10 @@ func (g *Generator) GenerateCode() (err error) {
 func completeFreeFormSchema(schema *openapi3.SchemaRef) {
 	v := schema.Value
 	if v.OneOf == nil && v.AnyOf == nil && v.AllOf == nil && v.Properties == nil {
-		if v.Type == "object" {
+		if v.Type == openapi3.TypeObject {
```
Comment (StevenLeiZhang) [inline pkg/definition/gen_sdk/gen_sdk.go:422]: If replace `v.Type == "object"` with `v.Type == openapi3.TypeObject`, how to handle `object`?

File: pkg/definition/gen_sdk/gen_sdk.go [hunk 2]
```diff
@@ -509,3 +547,63 @@ func newModifierOnLanguage(lang string, generator *Generator) Modifier {
 		panic("unsupported language: " + lang)
 	}
 }
+
+// getValueType returns the openapi type of the value
+func getValueType(i interface{}) CUEType {
```
Comment (StevenLeiZhang) [inline pkg/definition/gen_sdk/gen_sdk.go:552]: Shall we move this Function to Utils package?

File: pkg/definition/gen_sdk/gen_sdk.go [hunk 3]
```diff
@@ -461,12 +477,34 @@ func fixSchemaWithOneAnyAllOf(schema *openapi3.SchemaRef) {
 		}
 	}
 	schema.Value.Properties = nil
+
+	// remove duplicated type
+	for i, s := range oneOf {
+		if s.Value.Type != "" {
+			if _, ok := typeSet[s.Value.Type]; ok && s.Value.Type != openapi3.TypeObject {
+				duplicateIndex = append(duplicateIndex, i)
+				continue
```
[the remark this answers is not in the document]
Comment (chivalryq) [reply pkg/definition/gen_sdk/gen_sdk.go:486]: you're right.

File: pkg/definition/gen_sdk/gen_sdk.go (hunk 1 above)
Comment (chivalryq) [reply pkg/definition/gen_sdk/gen_sdk.go:422]: openapi3.TypeObject is "object" literally.

File: pkg/definition/gen_sdk/gen_sdk.go (hunk 2 above)
Comment (chivalryq) [reply pkg/definition/gen_sdk/gen_sdk.go:552]: This function is newly introduced. Let's hold it here until possible reuse occurs.

Comment (StevenLeiZhang) [reply pkg/definition/gen_sdk/gen_sdk.go:552]: > This function is newly introduced. Let's hold it here until possible reuse occurs.

ok

File: pkg/definition/gen_sdk/gen_sdk.go (hunk 1 above)
Comment (StevenLeiZhang) [reply pkg/definition/gen_sdk/gen_sdk.go:422]: > openapi3.TypeObject is "object" literally.

ok

**Review by StevenLeiZhang — APPROVED** [review]
LGTM

**Review by Somefive — APPROVED** [review]

**Pull request merged**